src/Core/SettingsChangesHistory.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <Core/Field.h>
+#include <IO/ReadHelpers.h>
+#include <boost/algorithm/string.hpp>
+#include <unordered_map>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+class ClickHouseVersion
+{
+public:
+    ClickHouseVersion(const String & version)
+    {
+        Strings split;
+        boost::split(split, version, [](char c){ return c == '.'; });
+        components.reserve(split.size());
+        for (const auto & split_element : split)
+        {
+            size_t component;
+            if (!tryParse(component, split_element))
+                throw Exception{ErrorCodes::BAD_ARGUMENTS, "Cannot parse ClickHouse version here: {}", version};
+            components.push_back(component);
+        }
+    }
+
+    ClickHouseVersion(const char * version) : ClickHouseVersion(String(version)) {}
+
+    bool operator<(const ClickHouseVersion & other) const
+    {
+        return components < other.components;
+    }
+
+private:
+    std::vector<size_t> components;
+};
+
+struct SettingChangesHistory
+{
+    struct Change
+    {
+        Change(const Field & value_, const ClickHouseVersion & version_) : value(value_), version(version_) {}
+
+        Field value;
+        ClickHouseVersion version;
+    };
+
+    SettingChangesHistory(const Field & initial_value_, const std::vector<Change> & changes_) : initial_value(initial_value_), changes(changes_) {}
+
+    Field getValueForVersion(const ClickHouseVersion & version) const
+    {
+        Field value = initial_value;
+        for (const auto & change : changes)
+        {
+            if (version < change.version)
+                return value;
+            value = change.value;
+        }
+        return value;
+    }
+
+    Field initial_value;
+    std::vector<Change> changes;
+};
+
+const std::unordered_map<String, SettingChangesHistory> settings_changes_history =
+{
+        {"enable_positional_arguments", {false, {{true, "22.7"}}}},
```

Comment:
this is gonna be fun to keep up to date. maybe add a comment where these settings are defined to make sure to update this?